File: PartialAbstraction.java
abstract class Shape{
    abstract double calculateArea();

    void displayArea() {
        System.out.println("Area: " + calculateArea());
    }
}

class Circle extends Shape {
    private double radius;

    Circle(double radius) {
        this.radius = radius;
    }
    double calculateArea() {
        return 3.14*radius*radius;
    }
}

class Rectangle extends Shape {
    private double length;
    private double width;

    Rectangle(double length, double width) {
        this.length = length;
        this.width = width;
    }

    double calculateArea() {
        return length*width;
    }
}

public class PartialAbstraction{
    public static void main(String[] args) {
        Circle circle = new Circle(5.0);
        Rectangle rectangle = new Rectangle(4.0, 6.0);

        System.out.print("Circle ");
        circle.displayArea();   
        System.out.print("Rectangle ");
        rectangle.displayArea(); 
    }
}

File: PartialAbstraction.cpp
#include <iostream>

using namespace std;

class Shape {
public:
    virtual double area()=0;

    void displayArea(){
        cout <<"Area: "<< area()<<endl;
    }
};

class Circle : public Shape {
private:
    double radius;
public:
    Circle(double r){
        radius=r;
    }

    double area(){
        return 3.14*radius*radius;
    }
};

class Rectangle : public Shape {
private:
    double length;
    double width;

public:
    Rectangle(double l, double w){
        length=l;
        width=w;
    }

    double area(){
        return length*width;
    }
};

int main() {
    Circle circle(5.0);
    Rectangle rectangle(4.0, 6.0);

    cout<<"Circle ";
    circle.displayArea(); 
    cout<<"Rectangle ";
    rectangle.displayArea();  

    return 0;
}